client/src/modules/list/components/ListItem.jsx

```diff
@@ -1,23 +1,120 @@
-import React, { PureComponent } from 'react';
+import React, { Fragment, PureComponent } from 'react';
 import classNames from 'classnames';
 import PropTypes from 'prop-types';
 
 import VotingBox from 'modules/list/components/VotingBox';
+import Textarea from 'common/components/Forms/Textarea';
+import TextInput from 'common/components/Forms/TextInput';
+import NewComment from '../../../common/components/Comments/NewComment';
+import Comment from '../../../common/components/Comments/Comment';
+import { ChevronDown, ChevronUp } from 'assets/images/icons';
 
 class ListItem extends PureComponent {
   constructor(props) {
     super(props);
     const { archived } = this.props;
-    this.state = { done: archived };
+    this.state = {
+      areDetailsVisible: false,
+      done: archived,
+      isHovered: false,
+      isNewCommentVisible: false
+    };
   }
 
   handleItemToggling = (authorName, id, archived) => () => {
     const { done } = this.state;
-    this.setState({ done: !done });
     const { toggleItem } = this.props;
+
+    this.setState({ done: !done });
     toggleItem(authorName, id, archived);
   };
 
+  toggleDetails = () =>
+    this.setState(({ areDetailsVisible }) => ({
+      areDetailsVisible: !areDetailsVisible
+    }));
+
+  showAddNewComment = () => this.setState({ isNewCommentVisible: true });
+
+  hideAddNewComment = () => this.setState({ isNewCommentVisible: false });
```

Comment:
Where hideAddNewComment is called? How to close form for adding new comment?

Reply:
escape press in new comment component

Reply:
ok. so there is no way to close item details via keyboard...